pkg/store/models/counterparty_test.go

```diff
@@ -249,3 +250,124 @@ func TestContactScan(t *testing.T) {
+			Expected: "http://www.example.com",
+		},
+		{
+			Input:    "http://localhost",
+			Expected: "http://localhost",
+		},
+		{
+			Input:    "http://127.0.0.1",
+			Expected: "http://127.0.0.1",
+		},
+		{
+			Input:    "http://[fd18:74e8:d33f:ffff::]",
+			Expected: "http://[fd18:74e8:d33f:ffff::]",
+		},
+		{
+			Input:    "gopher://username:password@the-weird-one.even_longer.example.website:12345/path/to/something/long?param1=123&secondparam=astringthistime",
+			Expected: "gopher://username:password@the-weird-one.even_longer.example.website:12345/path/to/something/long?param1=123&secondparam=astringthistime",
+		},
+	}
+
+	for idx, tc := range testcases {
+		t.Run(fmt.Sprintf("SuccessCase%d", idx), func(t *testing.T) {
+			//setup
+			counterparty := &models.Counterparty{}
+			counterparty.Website.String = tc.Input
+			if tc.Input != "" {
+				counterparty.Website.Valid = true
+			}
+
+			//test
+			website, err := counterparty.NormalizedWebsite()
+			require.NoError(t, err, "expected no error")
+			require.Equal(t, tc.Expected, website, "expected '%s', got '%s'")
+		})
+	}
+
+	t.Run("FailureIPV6NoSchema", func(t *testing.T) {
+		//setup
+		counterparty := &models.Counterparty{}
+		counterparty.Website.String = "[fd18:74e8:d33f:ffff::]"
+		counterparty.Website.Valid = true
+
+		//test
+		website, err := counterparty.NormalizedWebsite()
+		require.Error(t, err, "expected an error")
+		require.ErrorContains(t, err, "first path segment in URL cannot contain colon", "expected error to contain 'first path segment in URL cannot contain colon'")
+		require.Equal(t, website, "", "expected the empty string")
+	})
+
+	t.Run("ParsingFailureNull", func(t *testing.T) {
+		//setup
+		counterparty := &models.Counterparty{}
+		counterparty.Website.String = "\x00"
+		counterparty.Website.Valid = true
+
+		//test
+		website, err := counterparty.NormalizedWebsite()
+		require.Error(t, err, "expected an error")
+		require.ErrorContains(t, err, "invalid control character in URL", "expected error to contain 'invalid control character in URL'")
+		require.Equal(t, website, "", "expected the empty string")
+	})
+
+	t.Run("ParsingFailureHTTPSNull", func(t *testing.T) {
+		//setup
+		counterparty := &models.Counterparty{}
+		counterparty.Website.String = "https://\x00"
+		counterparty.Website.Valid = true
+
+		//test
+		website, err := counterparty.NormalizedWebsite()
+		require.Error(t, err, "expected an error")
+		require.ErrorContains(t, err, "invalid control character in URL", "expected error to contain 'invalid control character in URL'")
+		require.Equal(t, website, "", "expected the empty string")
+	})
+
+	t.Run("InvalidString", func(t *testing.T) {
+		//setup
+		counterparty := &models.Counterparty{}
+
+		//test
+		website, err := counterparty.NormalizedWebsite()
+		require.Error(t, err, "expected an error")
+		require.Equal(t, err, errors.ErrInvalidString, "expected ErrInvalidString")
+		require.Equal(t, website, "", "expected the empty string")
```

Comment:
This can be simplified as follows:

```suggestion
		require.ErrorIs(t, err, errors.ErrNullString)
		require.Empty(t, website, "empty string should be returned on null error.")
```

Reply:
Well, I've done it like this everywhere! I'll start using ErrorIs now